base/sparse/higherorderfns.jl

```diff
@@ -0,0 +1,889 @@
+                    args, ks, rows = _fusedupdatebc_all(rowsentinel, activerow, rows, defargs, ks, stopks, As)
+                    Cx = f(args...)
+                    activerow = min(rows...)
+                else
+                    Cx = defaultCx
+                end
+                Cx != fillvalue && (storedvals(C)[jo + Ci] = Cx)
+            end
+        end
+    end
+    return C
+end
+# helper method for broadcast/broadcast! methods just above
+@inline _expandsvert(C, A) = numrows(A) != numrows(C)
+@inline _expandsvert_all(C, ::Tuple{}) = ()
+@inline _expandsvert_all(C, As) = (_expandsvert(C, first(As)), _expandsvert_all(C, tail(As))...)
+@inline _expandshorz(C, A) = numcols(A) != numcols(C)
+@inline _expandshorz_all(C, ::Tuple{}) = ()
+@inline _expandshorz_all(C, As) = (_expandshorz(C, first(As)), _expandshorz_all(C, tail(As))...)
+@inline _startindforbccol(j, expandshorz, A) = expandshorz ? colstartind(A, 1) : colstartind(A, j)
+@inline _startindforbccol_all(j, ::Tuple{}, ::Tuple{}) = ()
+@inline _startindforbccol_all(j, expandshorzs, As) = (
+    _startindforbccol(j, first(expandshorzs), first(As)),
+    _startindforbccol_all(j, tail(expandshorzs), tail(As))...)
+@inline _stopindforbccol(j, expandshorz, A) = expandshorz ? colboundind(A, 1) : colboundind(A, j)
+@inline _stopindforbccol_all(j, ::Tuple{}, ::Tuple{}) = ()
+@inline _stopindforbccol_all(j, expandshorzs, As) = (
+    _stopindforbccol(j, first(expandshorzs), first(As)),
+    _stopindforbccol_all(j, tail(expandshorzs), tail(As))...)
+@inline _isemptycol(k, stopk) = k == stopk
+@inline _isemptycol_all(::Tuple{}, ::Tuple{}) = ()
+@inline _isemptycol_all(ks, stopks) = (
+    _isemptycol(first(ks), first(stopks)),
+    _isemptycol_all(tail(ks), tail(stopks))...)
+@inline _initrowforcol(j, rowsentinel, isempty, expandsvert, k, A) =
+    expandsvert || isempty ? convert(indtype(A), rowsentinel) : storedinds(A)[k]
+@inline _initrowforcol_all(j, rowsentinel, ::Tuple{}, ::Tuple{}, ::Tuple{}, ::Tuple{}) = ()
+@inline _initrowforcol_all(j, rowsentinel, isemptys, expandsverts, ks, As) = (
+    _initrowforcol(j, rowsentinel, first(isemptys), first(expandsverts), first(ks), first(As)),
+    _initrowforcol_all(j, rowsentinel, tail(isemptys), tail(expandsverts), tail(ks), tail(As))...)
+@inline _defargforcol(j, isempty, expandsvert, k, A) =
+    expandsvert && !isempty ? storedvals(A)[k] : zero(eltype(A))
+@inline _defargforcol_all(j, ::Tuple{}, ::Tuple{}, ::Tuple{}, ::Tuple{}) = ()
+@inline _defargforcol_all(j, isemptys, expandsverts, ks, As) = (
+    _defargforcol(j, first(isemptys), first(expandsverts), first(ks), first(As)),
+    _defargforcol_all(j, tail(isemptys), tail(expandsverts), tail(ks), tail(As))...)
+# fusing the following defs. avoids a few branches and construction of a tuple, yielding 1-20% runtime reduction
+# @inline _isactiverow(activerow, row) = row == activerow
+# @inline _isactiverow_all(activerow, ::Tuple{}) = ()
```

Comment:
The commented-out definitions are what I fused into `_fusedupdatebc` (for performance) per the comment on line 797. I left the unfused definitions in place (along with commented out calls to those methods before calls to `_fusedupdatebc`) as an easier-to-follow form of the fused version. Best!